liquid/stream.py

```diff
@@ -94,22 +100,19 @@ def next(self):
 			str: the next character
 		"""
 		ret = self.stream.read(1)
-		self.cursor = self.stream.tell()
 		return ret
 
 	def back(self):
 		"""
 		Put cursor 1-character back
 		"""
-		self.cursor -= 1
-		self.stream.seek(self.cursor)
+		self.stream.seek(self.cursor - 1)
```

Comment:
This could still be a problem when we try to retrieve the cursor from a Unicode character. Same as Line 126.

Reply:
I'm a little worried about using the length of the character because that's what caused the bug I saw in the first place.  I never isolated exactly what was the underlying cause, but it seemed that `StringIO` and `TextIOWrapper` had somewhat different behavior w/r/t seeking.

What if we got rid of the `back` method all together, and modified `until` to keep track of the last cursor as it moves through the stream, and then explicitly `seek` to that previous position?

I just pushed a commit that uses this technique, and it seems to pass all tests.

What do you think?

Reply:
Sounds great, let me take a look.